src/requirements/requirement-frontend-utils.ts

```diff
@@ -261,30 +261,45 @@ export function getMatchedRequirementFulfillmentSpecification(
 }
 
 const computeFulfillmentStatistics = (
+  requirementName: string,
   coursesTaken: readonly CourseTaken[],
+  overridenRequirementChoices: AppOverridenRequirementChoices,
   disallowTransferCredit: boolean,
   {
     fulfilledBy,
     eligibleCourses,
     perSlotMinCount,
+    slotNames,
     minNumberOfSlots,
   }: MatchedRequirementFulfillmentSpecificationBase
 ): RequirementFulfillmentStatisticsWithCourses => {
   const coursesThatFulfilledSubRequirements: CourseTaken[][] = eligibleCourses.map(() => []);
   const subRequirementProgress: number[] = eligibleCourses.map(() => 0);
   coursesTaken.forEach(courseTaken => {
-    const { courseId } = courseTaken;
-    if (!(disallowTransferCredit && courseIsAPIB(courseTaken))) {
+    const overrideOptions = overridenRequirementChoices[courseTaken.uniqueId];
+    const optInSlotNames = (overrideOptions && overrideOptions.optIn[requirementName]) || null;
+    const optOutSlotNames = (overrideOptions && overrideOptions.optOut[requirementName]) || null;
+
+    // block AP/IB equivalent courses if disallowTransferCredit
+    if (!(disallowTransferCredit && courseIsAPIB(courseTaken)) || optInSlotNames) {
       for (
         let subRequirementIndex = 0;
         subRequirementIndex < eligibleCourses.length;
         subRequirementIndex += 1
       ) {
-        if (
-          eligibleCourses[subRequirementIndex].includes(courseId) &&
-          subRequirementProgress[subRequirementIndex] < perSlotMinCount[subRequirementIndex]
+        const slotName = fulfilledBy === 'courses' ? slotNames[subRequirementIndex] : 'Course';
```

Comment:
Does this current infra support the case where I want to use my AP Physics C for a requirement with `checkerWarning` (i.e. Technical Electives)? 
![Screen Shot 2021-05-14 at 4 06 11 PM](https://user-images.githubusercontent.com/54298311/118323222-4e02c880-b4ce-11eb-82de-4774fe1d6af8.png)

Reply:
I don't see why not. Does it break when you test it out? It has a slotName "Course", so that'll need to be added as a slotName.

Reply:
Yes it does seem to work! My only remaining question then is do we have to know which slot (e.g. Course 3) that we want the AP/IB to apply to? 

When I just used slotName "Course" on Firestore, it seemed to work fine:
![Screen Shot 2021-05-15 at 4 19 07 PM](https://user-images.githubusercontent.com/54298311/118376995-46f4bc80-b599-11eb-8e8d-06a0604223f9.png)

Reply:
> Do we have to know which slot (e.g. Course 3) that we want the AP/IB to apply to?

We don't need to know (nor have control over) the rendered slot name, but we do need to know which data slot name we want the exam to apply to. The rendered slot name assignment order is automatic.

> When I just used slotName "Course" on Firestore, it seemed to work fine.

It actually won't work if we store it as "Course X", only "Course".

It's similar to reqs fulfilled by credits in that the rendered slot name "Course X" is not actually the data slot name. In this case, "Course" is the data slot name (so that's what should be stored on Firestore / needs to be modified if the data slot name changes).